clients/drcachesim/tests/record_filter_unit_tests.cpp

```diff
@@ -314,27 +317,46 @@ test_encodings2regdeps_filter()
         { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID, { 0x8 } }, true, { true } },
         // Encoding, modified by the record_filter encodings2regdeps.
-        { { TRACE_TYPE_ENCODING, 4, { 0xe78948 } }, true, { false } },
-        { { TRACE_TYPE_ENCODING, 8, { 0x0006090600010011 } }, false, { true } },
-        { { TRACE_TYPE_INSTR, 3, { 0x7f6fdd3ec360 } }, true, { true } },
-        // Same instr same chunk.
-        { { TRACE_TYPE_INSTR, 3, { 0x7f6fdd3ec360 } }, true, { true } },
-        // Unit header.
-        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP, { 0xc } }, true, { true } },
-        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID, { 0xd } }, true, { true } },
+        { { TRACE_TYPE_ENCODING, 3, { ENCODING_REAL_ISA } }, true, { false } },
+        { { TRACE_TYPE_ENCODING, 8, { ENCODING_REGDEPS_ISA } }, false, { true } },
+        { { TRACE_TYPE_INSTR, 3, { PC } }, true, { true } },
+        { { TRACE_TYPE_INSTR, 3, { PC } }, true, { true } },
+        { { TRACE_TYPE_INSTR, 3, { PC } }, true, { true } },
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CHUNK_FOOTER, { 0 } }, true, { true } },
+
+        // Chunk 2.
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_RECORD_ORDINAL, { 0xa } },
+          true,
+          { true } },
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP, { 0x7 } }, true, { true } },
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID, { 0x8 } }, true, { true } },
         // Duplicated encoding across chunk boundary.
-        { { TRACE_TYPE_ENCODING, 4, { 0xe78948 } }, true, { false } },
-        { { TRACE_TYPE_ENCODING, 8, { 0x0006090600010011 } }, false, { true } },
-        { { TRACE_TYPE_INSTR, 3, { 0x7f6fdd3ec360 } }, true, { true } },
-        // Unit header.
-        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP, { 0xe } }, true, { true } },
-        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID, { 0xf } }, true, { true } },
-        // Same instr in different chunk.
-        { { TRACE_TYPE_INSTR, 3, { 0x7f6fdd3ec360 } }, true, { true } },
+        { { TRACE_TYPE_ENCODING, 3, { ENCODING_REAL_ISA } }, true, { false } },
+        { { TRACE_TYPE_ENCODING, 8, { ENCODING_REGDEPS_ISA } }, false, { true } },
+        { { TRACE_TYPE_INSTR, 3, { PC } }, true, { true } },
+        { { TRACE_TYPE_INSTR, 3, { PC } }, true, { true } },
+        { { TRACE_TYPE_INSTR, 3, { PC } }, true, { true } },
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CHUNK_FOOTER, { 1 } }, true, { true } },
+
+        // Chunk 3.
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_RECORD_ORDINAL, { 0xe } },
+          true,
+          { true } },
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP, { 0x7 } }, true, { true } },
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID, { 0x8 } }, true, { true } },
+        // Same instr in different chunk. We expect record_filter to add
+        // ENCODING_REGDEPS_ISA here (not ENCODING_REAL_ISA).
```

Comment:
?  I don't understand this: where's the real encoding in the input?  Looks like you have an ill-formed input trace here with a missing encoding.  If that was the real intent, that has to be clearly commented: but I don't think that's the best way to go about testing.  To check changing chunks, add a trim filter or instruction removal filter or something to have the output be different from the input.

Reply:
Not my intent, that is indeed an ill-formed trace. I didn't realize `test_chunk_update()` was using a trim filter for that.
Fixed, and added a couple more test cases where the number of trace_entry_t records for encodings differ between real ISA and regdeps ISA.